Application/EdFi.Ods.AdminApi.AdminConsole/Features/WorkerInstances/ReadWorkerInstance.cs

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Linq;
+using System.Net.Http.Json;
+using System.Text.Json;
+using AutoMapper;
+using EdFi.Ods.AdminApi.AdminConsole.Features.Instances;
+using EdFi.Ods.AdminApi.AdminConsole.Infrastructure.DataAccess.Models;
+using EdFi.Ods.AdminApi.AdminConsole.Infrastructure.Services.Instances.Queries;
+using EdFi.Ods.AdminApi.Common.Features;
+using EdFi.Ods.AdminApi.Common.Infrastructure;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+
+namespace EdFi.Ods.AdminApi.AdminConsole.Features.WorkerInstances;
+
+public class ReadWorkerInstance : IFeature
+{
+    public void MapEndpoints(IEndpointRouteBuilder endpoints)
+    {
+        AdminApiEndpointBuilder.MapGet(endpoints, "/instances", GetInstances)
+            .BuildForVersions(AdminApiVersions.AdminConsole);
+
+        AdminApiEndpointBuilder.MapGet(endpoints, "/instances/{id}", GetInstanceById)
+            .WithRouteOptions(b => b.WithResponse<InstanceModel>(200))
+            .BuildForVersions(AdminApiVersions.AdminConsole);
+    }
+
+    internal async Task<IResult> GetInstances(IMapper mapper, [FromServices] IGetInstancesQuery getInstancesQuery, string? tenantName, string? status)
+    {
+        var instances = await getInstancesQuery.Execute(tenantName, status);
+        var instanceModels = mapper.Map<List<InstanceWorkerModel>>(instances);
+        if (instanceModels.Any())
+        {
+            return Results.Ok(instanceModels);
+        }
+        return Results.NotFound();
```

Comment:
In this scenario, where there are no instances it should probably return an empty result, instead of NotFound.